From 865cea40b69741c3da2574176876463233b2b67c Mon Sep 17 00:00:00 2001
From: Ram Amrani <Ram.Amrani@Cavium.com>
Date: Tue, 24 Jan 2017 13:50:34 +0200
Subject: [PATCH 130/130] RDMA/qedr: Return success when not changing QP state

If the user is requesting us to change the QP state to the same state
that it is already in, return success instead of failure.

Signed-off-by: Ram Amrani <Ram.Amrani@cavium.com>
Signed-off-by: Michal Kalderon <Michal.Kalderon@cavium.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
---
 drivers/infiniband/hw/qedr/verbs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/infiniband/hw/qedr/verbs.c b/drivers/infiniband/hw/qedr/verbs.c
index 84dcd73..27d90a8 100644
--- a/drivers/infiniband/hw/qedr/verbs.c
+++ b/drivers/infiniband/hw/qedr/verbs.c
@@ -1657,7 +1657,7 @@ static int qedr_update_qp_state(struct qedr_dev *dev,
	int status = 0;

	if (new_state == qp->state)
-		return 1;
+		return 0;

	switch (qp->state) {
	case QED_ROCE_QP_STATE_RESET:
--
1.8.3.1
